### PR TITLE
Eui code editor for stored layer config

### PR DESCRIPTION
**Pull request by corrigancd**
@szydan, @niknbr, having some difficulty with this. The reason being that the options UI has a different scope to the ETM vis controller. An issue with the current approach is that the apply changes button does not become an option when I change something in the storedLayerConfig component. I think this because I had to do some trickery to get data to the code editor component and to be able to update props in vis Controller. Any better way to do this would be great, 

storedLayerConfigNotTriggeringApplyChanges

### CODE REVIEW

File: public/visController.js [hunk 1]
```diff
@@ -74,7 +74,9 @@ define(function (require) {
     });
     //Using $root as mechanism to pass data to vis-editor-vis-options scope
     $scope.$root.etm = {
-      shapeFields: shapeFields
+      shapeFields: shapeFields,
+      getStoredLayerConfig: () => { return $scope.vis.params.storedLayerConfig; },
+      updateStoredLayerConfig: (newConfig) => $scope.vis.params.storedLayerConfig = newConfig
```
Comment (niknbr) [inline public/visController.js:79]: Would the button be enabled if we do this ?
```javascript
updateStoredLayerConfig: (newConfig) => {
   $scope.vis.params.storedLayerConfig = newConfig;
   $scope.$apply()
}
```

Comment (corrigancd) [reply public/visController.js:79]: Thanks, unfortunately, it remains not enabled. I'm getting that the visController scope and the vis scope are separate. So there somehow needs to be a more definite link between them, will keep looking into it!

**Comment by corrigancd** [comment]
Ok, second attempt: 
StoredLayerCodeEditorComponent

File: public/directives/storedLayerConfig.js [hunk 2]
```diff
@@ -1,26 +1,52 @@
...
+        function updateCodeEditor() {
+          render(
+            <EuiCodeEditor
+              mode="javascript"
+              theme="github"
+              width="100%"
+              tabSize={2}
+              value={$scope.config}
+              onChange={(newConfig) => updateConfig(newConfig)}
+              setOptions={{
```
Comment (szydan) [inline public/directives/storedLayerConfig.js:42]: can you add the 
tabSize: 2

so the json is indented a bit

File: public/options.html [hunk 3]
```diff
@@ -282,9 +282,14 @@
...
   <div uib-accordion-group class="panel-default" heading="Stored Layer Config">
+    <div class="form-group">
+      <label class="etm-options-section">
+        Stored Layer Config <kbn-info info="Config object for Stored Layers. The last item in the array is a default object, containing all default config parameters. Other configuration items must contain a spatial_path attribute."></kbn-info>
+      </label>
 
     <stored-layer-config config="vis.params.storedLayerConfig"></stored-layer-config>
     
   </div>
+  </div>
```
Comment (szydan) [inline public/options.html:293]: I think the div above should be indented a bit

File: public/directives/storedLayerConfig.js (hunk 2 above)
Comment (corrigancd) [reply public/directives/storedLayerConfig.js:42]: So tabSize means that when a tab input is received, the indent will be 2 spaces, this is currently implemented. 

I opted not to do something like JSON.stringify(json.parse(config), null, 2) on the string input as it parses array arrays from:
['example array']
TO
[
    'example array'
]

File: public/options.html (hunk 3 above)
Comment (corrigancd) [reply public/options.html:293]: done

**Pull request merged**